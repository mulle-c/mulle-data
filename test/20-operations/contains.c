#include <mulle-data/mulle-data.h>

#include <stdio.h>

static void  mulle_range_print( struct mulle_range a)
{
   if( ! a.length)
      printf( "NULL");  
   else
      printf( "%td-%td (%td)", a.location, a.location + a.length - 1, a.length);  
}


int   main( int argc, char *argv[])
{
   struct mulle_range   a;
   struct mulle_range   b;
   unsigned int         i, n;
   int                  flag;

   for( a.location = 0; a.location < 4; a.location += 1)
      for( b.location = 0; b.location < 4; b.location += 1)
         for( a.length = 0; a.length < 4; a.length += 1)
            for( b.length = 0; b.length < 4; b.length += 1)
            {
               flag = mulle_range_contains( a, b);
               mulle_range_print( a);
               printf( " contains ");
               mulle_range_print( b);
               printf( " = %s\n", flag ? "YES" : "NO");
            }

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
