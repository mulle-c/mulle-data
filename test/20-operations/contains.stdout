NULL contains NULL = NO
NULL contains 0-0 (1) = NO
NULL contains 0-1 (2) = NO
NULL contains 0-2 (3) = NO
0-0 (1) contains NULL = YES
0-0 (1) contains 0-0 (1) = YES
0-0 (1) contains 0-1 (2) = NO
0-0 (1) contains 0-2 (3) = NO
0-1 (2) contains NULL = YES
0-1 (2) contains 0-0 (1) = YES
0-1 (2) contains 0-1 (2) = YES
0-1 (2) contains 0-2 (3) = NO
0-2 (3) contains NULL = YES
0-2 (3) contains 0-0 (1) = YES
0-2 (3) contains 0-1 (2) = YES
0-2 (3) contains 0-2 (3) = YES
NULL contains NULL = NO
NULL contains 1-1 (1) = NO
NULL contains 1-2 (2) = NO
NULL contains 1-3 (3) = NO
0-0 (1) contains NULL = NO
0-0 (1) contains 1-1 (1) = NO
0-0 (1) contains 1-2 (2) = NO
0-0 (1) contains 1-3 (3) = NO
0-1 (2) contains NULL = YES
0-1 (2) contains 1-1 (1) = YES
0-1 (2) contains 1-2 (2) = NO
0-1 (2) contains 1-3 (3) = NO
0-2 (3) contains NULL = YES
0-2 (3) contains 1-1 (1) = YES
0-2 (3) contains 1-2 (2) = YES
0-2 (3) contains 1-3 (3) = NO
NULL contains NULL = NO
NULL contains 2-2 (1) = NO
NULL contains 2-3 (2) = NO
NULL contains 2-4 (3) = NO
0-0 (1) contains NULL = NO
0-0 (1) contains 2-2 (1) = NO
0-0 (1) contains 2-3 (2) = NO
0-0 (1) contains 2-4 (3) = NO
0-1 (2) contains NULL = NO
0-1 (2) contains 2-2 (1) = NO
0-1 (2) contains 2-3 (2) = NO
0-1 (2) contains 2-4 (3) = NO
0-2 (3) contains NULL = YES
0-2 (3) contains 2-2 (1) = YES
0-2 (3) contains 2-3 (2) = NO
0-2 (3) contains 2-4 (3) = NO
NULL contains NULL = NO
NULL contains 3-3 (1) = NO
NULL contains 3-4 (2) = NO
NULL contains 3-5 (3) = NO
0-0 (1) contains NULL = NO
0-0 (1) contains 3-3 (1) = NO
0-0 (1) contains 3-4 (2) = NO
0-0 (1) contains 3-5 (3) = NO
0-1 (2) contains NULL = NO
0-1 (2) contains 3-3 (1) = NO
0-1 (2) contains 3-4 (2) = NO
0-1 (2) contains 3-5 (3) = NO
0-2 (3) contains NULL = NO
0-2 (3) contains 3-3 (1) = NO
0-2 (3) contains 3-4 (2) = NO
0-2 (3) contains 3-5 (3) = NO
NULL contains NULL = NO
NULL contains 0-0 (1) = NO
NULL contains 0-1 (2) = NO
NULL contains 0-2 (3) = NO
1-1 (1) contains NULL = NO
1-1 (1) contains 0-0 (1) = NO
1-1 (1) contains 0-1 (2) = NO
1-1 (1) contains 0-2 (3) = NO
1-2 (2) contains NULL = NO
1-2 (2) contains 0-0 (1) = NO
1-2 (2) contains 0-1 (2) = NO
1-2 (2) contains 0-2 (3) = NO
1-3 (3) contains NULL = NO
1-3 (3) contains 0-0 (1) = NO
1-3 (3) contains 0-1 (2) = NO
1-3 (3) contains 0-2 (3) = NO
NULL contains NULL = NO
NULL contains 1-1 (1) = NO
NULL contains 1-2 (2) = NO
NULL contains 1-3 (3) = NO
1-1 (1) contains NULL = YES
1-1 (1) contains 1-1 (1) = YES
1-1 (1) contains 1-2 (2) = NO
1-1 (1) contains 1-3 (3) = NO
1-2 (2) contains NULL = YES
1-2 (2) contains 1-1 (1) = YES
1-2 (2) contains 1-2 (2) = YES
1-2 (2) contains 1-3 (3) = NO
1-3 (3) contains NULL = YES
1-3 (3) contains 1-1 (1) = YES
1-3 (3) contains 1-2 (2) = YES
1-3 (3) contains 1-3 (3) = YES
NULL contains NULL = NO
NULL contains 2-2 (1) = NO
NULL contains 2-3 (2) = NO
NULL contains 2-4 (3) = NO
1-1 (1) contains NULL = NO
1-1 (1) contains 2-2 (1) = NO
1-1 (1) contains 2-3 (2) = NO
1-1 (1) contains 2-4 (3) = NO
1-2 (2) contains NULL = YES
1-2 (2) contains 2-2 (1) = YES
1-2 (2) contains 2-3 (2) = NO
1-2 (2) contains 2-4 (3) = NO
1-3 (3) contains NULL = YES
1-3 (3) contains 2-2 (1) = YES
1-3 (3) contains 2-3 (2) = YES
1-3 (3) contains 2-4 (3) = NO
NULL contains NULL = NO
NULL contains 3-3 (1) = NO
NULL contains 3-4 (2) = NO
NULL contains 3-5 (3) = NO
1-1 (1) contains NULL = NO
1-1 (1) contains 3-3 (1) = NO
1-1 (1) contains 3-4 (2) = NO
1-1 (1) contains 3-5 (3) = NO
1-2 (2) contains NULL = NO
1-2 (2) contains 3-3 (1) = NO
1-2 (2) contains 3-4 (2) = NO
1-2 (2) contains 3-5 (3) = NO
1-3 (3) contains NULL = YES
1-3 (3) contains 3-3 (1) = YES
1-3 (3) contains 3-4 (2) = NO
1-3 (3) contains 3-5 (3) = NO
NULL contains NULL = NO
NULL contains 0-0 (1) = NO
NULL contains 0-1 (2) = NO
NULL contains 0-2 (3) = NO
2-2 (1) contains NULL = NO
2-2 (1) contains 0-0 (1) = NO
2-2 (1) contains 0-1 (2) = NO
2-2 (1) contains 0-2 (3) = NO
2-3 (2) contains NULL = NO
2-3 (2) contains 0-0 (1) = NO
2-3 (2) contains 0-1 (2) = NO
2-3 (2) contains 0-2 (3) = NO
2-4 (3) contains NULL = NO
2-4 (3) contains 0-0 (1) = NO
2-4 (3) contains 0-1 (2) = NO
2-4 (3) contains 0-2 (3) = NO
NULL contains NULL = NO
NULL contains 1-1 (1) = NO
NULL contains 1-2 (2) = NO
NULL contains 1-3 (3) = NO
2-2 (1) contains NULL = NO
2-2 (1) contains 1-1 (1) = NO
2-2 (1) contains 1-2 (2) = NO
2-2 (1) contains 1-3 (3) = NO
2-3 (2) contains NULL = NO
2-3 (2) contains 1-1 (1) = NO
2-3 (2) contains 1-2 (2) = NO
2-3 (2) contains 1-3 (3) = NO
2-4 (3) contains NULL = NO
2-4 (3) contains 1-1 (1) = NO
2-4 (3) contains 1-2 (2) = NO
2-4 (3) contains 1-3 (3) = NO
NULL contains NULL = NO
NULL contains 2-2 (1) = NO
NULL contains 2-3 (2) = NO
NULL contains 2-4 (3) = NO
2-2 (1) contains NULL = YES
2-2 (1) contains 2-2 (1) = YES
2-2 (1) contains 2-3 (2) = NO
2-2 (1) contains 2-4 (3) = NO
2-3 (2) contains NULL = YES
2-3 (2) contains 2-2 (1) = YES
2-3 (2) contains 2-3 (2) = YES
2-3 (2) contains 2-4 (3) = NO
2-4 (3) contains NULL = YES
2-4 (3) contains 2-2 (1) = YES
2-4 (3) contains 2-3 (2) = YES
2-4 (3) contains 2-4 (3) = YES
NULL contains NULL = NO
NULL contains 3-3 (1) = NO
NULL contains 3-4 (2) = NO
NULL contains 3-5 (3) = NO
2-2 (1) contains NULL = NO
2-2 (1) contains 3-3 (1) = NO
2-2 (1) contains 3-4 (2) = NO
2-2 (1) contains 3-5 (3) = NO
2-3 (2) contains NULL = YES
2-3 (2) contains 3-3 (1) = YES
2-3 (2) contains 3-4 (2) = NO
2-3 (2) contains 3-5 (3) = NO
2-4 (3) contains NULL = YES
2-4 (3) contains 3-3 (1) = YES
2-4 (3) contains 3-4 (2) = YES
2-4 (3) contains 3-5 (3) = NO
NULL contains NULL = NO
NULL contains 0-0 (1) = NO
NULL contains 0-1 (2) = NO
NULL contains 0-2 (3) = NO
3-3 (1) contains NULL = NO
3-3 (1) contains 0-0 (1) = NO
3-3 (1) contains 0-1 (2) = NO
3-3 (1) contains 0-2 (3) = NO
3-4 (2) contains NULL = NO
3-4 (2) contains 0-0 (1) = NO
3-4 (2) contains 0-1 (2) = NO
3-4 (2) contains 0-2 (3) = NO
3-5 (3) contains NULL = NO
3-5 (3) contains 0-0 (1) = NO
3-5 (3) contains 0-1 (2) = NO
3-5 (3) contains 0-2 (3) = NO
NULL contains NULL = NO
NULL contains 1-1 (1) = NO
NULL contains 1-2 (2) = NO
NULL contains 1-3 (3) = NO
3-3 (1) contains NULL = NO
3-3 (1) contains 1-1 (1) = NO
3-3 (1) contains 1-2 (2) = NO
3-3 (1) contains 1-3 (3) = NO
3-4 (2) contains NULL = NO
3-4 (2) contains 1-1 (1) = NO
3-4 (2) contains 1-2 (2) = NO
3-4 (2) contains 1-3 (3) = NO
3-5 (3) contains NULL = NO
3-5 (3) contains 1-1 (1) = NO
3-5 (3) contains 1-2 (2) = NO
3-5 (3) contains 1-3 (3) = NO
NULL contains NULL = NO
NULL contains 2-2 (1) = NO
NULL contains 2-3 (2) = NO
NULL contains 2-4 (3) = NO
3-3 (1) contains NULL = NO
3-3 (1) contains 2-2 (1) = NO
3-3 (1) contains 2-3 (2) = NO
3-3 (1) contains 2-4 (3) = NO
3-4 (2) contains NULL = NO
3-4 (2) contains 2-2 (1) = NO
3-4 (2) contains 2-3 (2) = NO
3-4 (2) contains 2-4 (3) = NO
3-5 (3) contains NULL = NO
3-5 (3) contains 2-2 (1) = NO
3-5 (3) contains 2-3 (2) = NO
3-5 (3) contains 2-4 (3) = NO
NULL contains NULL = NO
NULL contains 3-3 (1) = NO
NULL contains 3-4 (2) = NO
NULL contains 3-5 (3) = NO
3-3 (1) contains NULL = YES
3-3 (1) contains 3-3 (1) = YES
3-3 (1) contains 3-4 (2) = NO
3-3 (1) contains 3-5 (3) = NO
3-4 (2) contains NULL = YES
3-4 (2) contains 3-3 (1) = YES
3-4 (2) contains 3-4 (2) = YES
3-4 (2) contains 3-5 (3) = NO
3-5 (3) contains NULL = YES
3-5 (3) contains 3-3 (1) = YES
3-5 (3) contains 3-4 (2) = YES
3-5 (3) contains 3-5 (3) = YES
